plugin/trino-hive/src/main/java/io/trino/plugin/hive/ion/IonPageSourceFactory.java

```diff
@@ -80,6 +85,9 @@ public Optional<ReaderPageSource> createPageSource(
             boolean originalFile,
             AcidTransaction transaction)
     {
+        if (!this.nativeTrinoEnabled) {
+            throw new TrinoException(HIVE_UNSUPPORTED_FORMAT, "Ion native trino integration is not enabled");
```

Comment:
The contract here is to return empty when the page source doesn't support the combination of format and options. 

The way this works is sort of odd (I assume for reasons of how it always worked in hive), basically it's not a direct mapping from file type to SerDe. There is a set of SerDes and they are effectively round-robbin'ed to find one that will work for that file type.

So returning empty here will allow folks who have the existing Ion-hive-serde (like Athena) to have that just get picked up instead.

So there are likely some cases we would want to throw a hard error, but this is still in what we might consider the "PageSourceFactory selection" phase.

Reply:
I see. I will just return an `Optional.empty()` here that would allow user to either fall to a default(like ion-hive-serde) or throw an error.